src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -294,6 +293,14 @@ private SelectItem GenerateSelectItem(SelectTermToken tokenIn)
 
             IEdmTypeReference elementTypeReference = targetElementType.ToTypeReference();
 
+            // When Creating Range Variables, we only need a Navigation Source when the elementTypeReference is a StructuredTypeReference.

```

Comment:
>When Creating Range Variables, we only need a Navigation Source when the elementTypeReference is a StructuredTypeReference. [](start = 15, length = 123)

What happen if we set the navigation source if the elementTypeReference is not a structuredTypeReference?

Reply:
Tests fail due to this https://github.com/OData/odata.net/blob/3f62d72c9d909dde519f130ffd8411f31ee95ca0/src/Microsoft.OData.Core/UriParser/Parsers/NodeFactory.cs#L63